internal/gcp/tags_test.go

```diff
@@ -0,0 +1,235 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+	"cloud.google.com/go/resourcemanager/apiv3/resourcemanagerpb"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const bufSize = 1024 * 1024
+
+func bufDialer(lis *bufconn.Listener) (net.Conn, error) {
+	return lis.Dial()
+}
+
+type fakeTagKeysServer struct {
+	resourcemanagerpb.UnimplementedTagKeysServer
+}
+
+func (s *fakeTagKeysServer) GetNamespacedTagKey(ctx context.Context, req *resourcemanagerpb.GetNamespacedTagKeyRequest) (*resourcemanagerpb.TagKey, error) {
+	if req.Name == "test-project/existing-key" {
+		return &resourcemanagerpb.TagKey{
+			Name:      "projects/test-project/existing-key",
+			ShortName: "existing-key",
+		}, nil
+	}
+	return nil, fmt.Errorf("tag key not found")
+}
+
+type fakeTagValuesServer struct {
+	resourcemanagerpb.UnimplementedTagValuesServer
+}
+
+func (s *fakeTagValuesServer) GetNamespacedTagValue(ctx context.Context, req *resourcemanagerpb.GetNamespacedTagValueRequest) (*resourcemanagerpb.TagValue, error) {
+	if req.Name == "test-project/existing-key/existing-value" {
+		return &resourcemanagerpb.TagValue{
+			Name:      "projects/test-project/existing-key/existing-value",
+			ShortName: "existing-value",
+		}, nil
+	}
+	return nil, fmt.Errorf("tag value not found")
+}
+
+func TestLookupKeyWithFakeGRPCServer(t *testing.T) {
+	lis := bufconn.Listen(bufSize)
+
+	s := grpc.NewServer()
+	resourcemanagerpb.RegisterTagKeysServer(s, &fakeTagKeysServer{})
+
+	go func() {
+		if err := s.Serve(lis); err != nil && err != grpc.ErrServerStopped {
+			t.Errorf("Server exited with error: %v", err)
+		}
+	}()
+	defer s.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(func(ctx context.Context, s string) (net.Conn, error) {
+		return bufDialer(lis)
+	}), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("Failed to dial bufnet: %v", err)
+	}
+	defer conn.Close()
+
+	keysClient, err := resourcemanager.NewTagKeysClient(ctx, option.WithGRPCConn(conn))
+	if err != nil {
+		t.Fatalf("Failed to create TagKeysClient: %v", err)
+	}
+
+	mgr := NewTagsManager(keysClient, nil)
+
+	key, err := mgr.LookupKey(ctx, "test-project", "existing-key")
+	if err != nil {
+		t.Fatalf("LookupKey failed: %v", err)
+	}
+
+	if key.Name != "projects/test-project/existing-key" {
+		t.Errorf("Expected key name 'projects/test-project/existing-key', got: %v", key.Name)
+	}
```

Comment:
I would lean towards using testify assertions here like you did in your other tests for consistency, but since ultimately both works you can consider this comment optional

(the same for all the other assertions in this file)